### DISCUSSION TITLE
How to access child properties inside parent component?

### DISCUSSION THREAD

**Discussion by milanchheda** [livewire/livewire#3267] [1. Help] [answered]
Hi,

I have a simple parent component with a child component. 

Parent Component:
```
class ParentComponent extends Component {
    public $messsage;

   public function render()
    {
        return view('livewire.parent');
    }

    public function submit()
        {
            // I want to access child properties here 
            // so that I can process $message based on the values of child properties
            return view('livewire.parent');
        }
}
```

```
class ChildComponent extends Component {
    $public $filterValues [
        'start' => true,
        'category' => 3,
        'class' => 4
        ];


   public function render()
    {
        return view('livewire.child');
    }

}
```

When the `submit` method is called via `form` submit using `wire:submit.prevent="submit"`, how can I access child properties inside `submit` method of parent?

**Comment by joshhanley** [comment, marked as answer]
@milanchheda the short answer is, you can't.

You can't access data across components. If you need to keep data in sync between a child and parent you need to use events to communicate between them. See here in the docs https://laravel-livewire.com/docs/2.x/events

I would generally recommend not splitting into multiple Livewire components unless you really need to, as keep state in sync between two components using events can get messy. Instead make use of blade components for splitting up of your HTML.

Hope this helps!

**Reply by milanchheda** [reply]
Thanks @joshhanley. I think, it makes sense and I would rather refactor the code than syncing properties across components using events.

**Reply by joshhanley** [reply]
No worries 🙂
